Toggl.iOS/ViewControllers/Common/SelectorViewController.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Toggl.Core.UI.Views;
+using Toggl.iOS.Extensions;
+using Toggl.iOS.ViewSources.Common;
+using UIKit;
+
+namespace Toggl.iOS.ViewControllers.Common
+{
+    public sealed class SelectorViewController<T> : UIViewController
+    {
+        private readonly ImmutableList<SelectOption<T>> options;
+        private readonly string title;
+        private readonly Action<T> onChosen;
+        private int selectedIndex;
+
+        private UITableView tableView;
+        private SelectorTableViewSource<T> source;
+
+        public SelectorViewController(
+            string title,
+            IEnumerable<SelectOption<T>> options,
+            int initialIndex,
+            Action<T> onChosen)
+        {
+            selectedIndex = initialIndex;
+            this.title = title;
+            this.options = options.ToImmutableList();
+            this.onChosen = onChosen;
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+
+            Title = title;
+
+            tableView = new UITableView(View.Bounds);
+            tableView.TableFooterView = new UIView();
+            View.AddSubview(tableView);
+
+            source = new SelectorTableViewSource<T>(tableView, options, selectedIndex, onItemSelected);
+
+            tableView.Source = source;
+        }
+
+        public override void ViewWillDisappear(bool animated)
+        {
+            base.ViewWillDisappear(animated);
+
+            var selected = options[selectedIndex];
+            onChosen(selected.Item);
+        }
+
+        private void onItemSelected(int selectedIndex)
+        {
+            this.selectedIndex = selectedIndex;
+
+            if (NavigationController != null)
```

Comment:
Can we drop the `{}`s?
```suggestion:0+8
            if (NavigationController != null)
                NavigationController.PopViewController(true);
            else
                this.Dismiss();
```

Reply:
Mmmm... No? 😄

Seriously, we should write guidelines on this. I do like my `{}` even for single statements, it prevents mistakes when adding more lines.

Reply:
When `if` has an `else`, I am also in favor of putting stuff into the `{}`. Although I think that mistake prevention is a non-issue. Everybody uses that argument, but we simply don't change code that caressly. Besides, the IDE will format your code in such a way that you will see the incorrect indentation anyway. 🤷‍♂

Reply:
There are two things I don't trust:
- Developers care (even my own, we all have those days)
- IDEs (mostly since I do Xamarin)

;)